### DISCUSSION TITLE
Как сделать схожий конфиг как ByeDPIAndroid в zapret?

### DISCUSSION THREAD

**Discussion by SpiritOTHawk** [bol-van/zapret#472] [Q&A]
Сделал на рандом настройки и заработал дискорд чат + войс.
![Screenshot_20241010-110418](https://github.com/user-attachments/assets/68ddd137-3fd6-4c65-9541-cf7d336d11fa)
![Screenshot_20241010-110428](https://github.com/user-attachments/assets/780459ca-6847-487b-94eb-9ad8d912d600)

**Comment by rolliebae** [comment]
Ваш параметр `NFQWS_OPT_DESYNC` содержит сложную и детализированную настройку для обхода DPI, с использованием множества опций для разных хостов и методов desync. Давайте разберем, что делает каждая часть этого параметра:

### Общий вид:
```bash
NFQWS_OPT_DESYNC="--hostlist=/opt/zapret/ipset/youtube.txt --dpi-desync=fake,split --dpi-desync-ttl=5 --dpi-desync-fake-tls=0x00000000 --dpi-desync-split-pos=1 --new --hostlist=/opt/zapret/ipset/googlevideo.txt --dpi-desync=fake,split2 --dpi-desync-fooling=badseq --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --dpi-desync-split-pos=1 --new --dpi-desync=fake --dpi-desync-ttl=4"
```

### Пояснение каждой части:

1. **`--hostlist=/opt/zapret/ipset/youtube.txt`**
   - Указывает файл с доменами или IP-адресами, к которым будут применяться следующие параметры. В данном случае это `youtube.txt`, содержащий список хостов, относящихся к YouTube.

2. **`--dpi-desync=fake,split`**
   - Используется метод DPI desync: комбинация методов `fake` и `split`. Метод `fake` отправляет поддельный пакет, а `split` разбивает пакеты.

3. **`--dpi-desync-ttl=5`**
   - Устанавливает TTL (Time to Live) для пакетов desync на 5.

4. **`--dpi-desync-fake-tls=0x00000000`**
   - Использует поддельное TLS-сообщение. Здесь указано значение `0x00000000`, что может означать отправку пустого или минимально изменённого TLS-сообщения.

5. **`--dpi-desync-split-pos=1`**
   - Установка позиции разбиения пакетов на 1 байт.

6. **`--new`**
   - Указывает, что параметры, следующие за этим флагом, применяются к новому набору хостов или соединений.

7. **`--hostlist=/opt/zapret/ipset/googlevideo.txt`**
   - Ещё один файл с хостами, на этот раз относящийся к `googlevideo.txt` (Google Video, обычно используется для видео на YouTube).

8. **`--dpi-desync=fake,split2`**
   - Используется метод DPI desync: комбинация `fake` и `split2`. Метод `split2` немного отличается от `split`, возможно, другой логикой разбиения.

9. **`--dpi-desync-fooling=badseq`**
   - Использует метод `badseq` для подделки последовательности пакетов, что может сбить DPI.

10. **`--dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin`**
    - Использует заранее подготовленный файл с поддельным TLS-сообщением ClientHello для хоста `www.google.com`.

11. **`--dpi-desync-split-pos=1`**
    - Установка позиции разбиения пакетов на 1 байт (снова для хостов из `googlevideo.txt`).

12. **`--new`**
    - Указывает начало новой группы настроек.

13. **`--dpi-desync=fake`**
    - Применяет метод DPI desync: только `fake`. Это означает, что отправляются только поддельные пакеты без разбиения.

14. **`--dpi-desync-ttl=4`**
    - Устанавливает TTL на 4 для последней группы параметров.

### Как это работает:
Эти параметры настраивают различные методы обхода DPI для разных доменов (в данном случае, YouTube и Google Video). В зависимости от хоста, используются разные комбинации методов `fake`, `split`, `split2`, `badseq`, а также различные TTL и поддельные TLS-сообщения.

### Пример упрощённой конфигурации:
Если вы хотите создать конфигурацию с похожим функционалом, но более простой, например, для всех хостов, вы можете использовать что-то вроде:

```bash
NFQWS_OPT_DESYNC="--dpi-desync=fake,split --dpi-desync-ttl=5 --dpi-desync-fake-tls=0x00000000 --dpi-desync-split-pos=1"
```

Если вы хотите создать конфигурацию на основе настроек из **ByeDPIAndroid**, которые были показаны на скриншоте, то можно адаптировать параметры, чтобы они соответствовали тем, которые указаны в конфигурации **ByeDPIAndroid**. Учитывая параметры, которые были на скриншоте, можно привести пример конфигурации для **zapret**, аналогичной **ByeDPIAndroid**:

### 1. **Desync method (Disorder)**:
   - В **ByeDPIAndroid** выбран метод разсинхронизации **Disorder** — это аналог метода **fake** и **split** в zapret.
   - Скомбинируем методы `fake` и `split` для достижения аналогичного эффекта. В zapret это может выглядеть так:
   
   ```bash
   --dpi-desync=fake,split
   ```

### 2. **Split position (3)**:
   - В **ByeDPIAndroid** используется позиция разбиения `3`. В zapret можно задать это через параметр `--dpi-desync-split-pos=3`:
   
   ```bash
   --dpi-desync-split-pos=3
   ```

### 3. **Split at host**:
   - В **ByeDPIAndroid** включена опция разбиения пакета на уровне хоста. В **zapret** это можно включить с помощью `--split-http-host`:
   
   ```bash
   --split-http-host
   ```

### 4. **Drop SACK**:
   - В **ByeDPIAndroid** активирована опция **Drop SACK** (Selective Acknowledgment). В **zapret** это можно включить с помощью параметра `--drop-sack`:
   
   ```bash
   --drop-sack
   ```

### 5. **Desync для HTTP, HTTPS и UDP**:
   - В **ByeDPIAndroid** включены опции для разсинхронизации трафика HTTP, HTTPS и UDP. В **zapret** это можно задать с помощью следующих параметров:
   
   ```bash
   --desync-http --desync-https --desync-udp
   ```

### 6. **Host mixed case, Domain mixed case, Host remove spaces**:
   - В **ByeDPIAndroid** включены опции для изменения регистра (mixed case) и удаления пробелов. В **zapret** это можно сделать с помощью следующих параметров:
   
   ```bash
   --host-mixed-case
   --domain-mixed-case
   --host-remove-spaces
   ```

### 7. **TTL (Default TTL = 0)**:
   - В **ByeDPIAndroid** TTL установлен на `0` по умолчанию. В запрете можно настроить TTL через параметр `--dpi-desync-ttl`. В вашем случае это может быть `0`, однако, если требуется другой TTL, можно изменить его, например:

   ```bash
   --dpi-desync-ttl=0
   ```

### Итоговая конфигурация для zapret:
На основе этих параметров и конфигурации **ByeDPIAndroid**, итоговый пример конфигурации может выглядеть так:

```bash
NFQWS_OPT_DESYNC="--dpi-desync=fake,split --dpi-desync-split-pos=3 --split-http-host --drop-sack --desync-http --desync-https --desync-udp --host-mixed-case --domain-mixed-case --host-remove-spaces --dpi-desync-ttl=0"
```

### Дополнительные параметры:
Если требуется добавить дополнительные параметры, такие как фильтрация по доменам (например, для YouTube или GoogleVideo), вы можете добавить их с помощью `--hostlist` и указать соответствующие списки доменов:

```bash
--hostlist=/opt/zapret/ipset/youtube.txt
```

### Заключение:
Эта конфигурация будет максимально приближена к тому, что используется в **ByeDPIAndroid**, и поможет настроить **zapret** так, чтобы он работал как аналогичный инструмент для обхода DPI.

**Reply by Bzdyshek** [reply]
А какому набору в запрете соответствует disordered out-of-band, данные OOB = а?
